### PR TITLE
Move X.509 certificate tests under S.S.Cryptography

### PR DESCRIPTION
**Pull request by vcsjones**
Contributes to #66338

### CODE REVIEW

**Comment by bartonjs** [comment]
I think there are some contexts that disable/skip the library outright, like wasm, so once you think you're ready it probably needs an extraplatforms run to see how many places start exploding.

**Comment by vcsjones** [comment]
@bartonjs bleh, you're right:

https://github.com/dotnet/runtime/blob/91004ed8ffcfa9513d5927aa29d00b7169d3b3cb/src/libraries/tests.proj#L374

What would be your preference?

1. Modify the S.S.Cryptography.Tests.csproj to not include the `<Compile .. />` items if the target is browser for our moved files.
2. Throw `[ActiveIssue("https://github.com/dotnet/runtime/issues/37669", TestPlatforms.Browser)]` on everything in under `\X509Certificates`?

**Comment by bartonjs** [comment]
Probably (3): `[SkipOnPlatform(Browser)]` on all of the classes that actually fail or need certificates.  Some of the types (SANBuilder, X500DNBuilder, most of the extension types) probably just work and don't get tested.

Since X509 on wasm hasn't ever been in the plans calling it 37669 feels wrong to me.

**Comment by vcsjones** [comment]
> most of the extension types

Unfortunately many of these try to go through an X509Pal which WASM doesn't have set up - not even the managed one.

We can take a follow up action after this PR to implement as much of an IX509Pal as we can with the managed encoding and decoding.

**Comment by vcsjones** [comment]
I'm going to optimistically mark this one ready for review since some of the WASM runs are passing now.

File: src/libraries/System.Security.Cryptography/tests/X509Certificates/RevocationTests/AiaTests.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace System.Security.Cryptography.X509Certificates.Tests.RevocationTests
 {
     [ActiveIssue("https://github.com/dotnet/runtime/issues/57506", typeof(PlatformDetection), nameof(PlatformDetection.IsMonoRuntime), nameof(PlatformDetection.IsMariner))]
-    [SkipOnPlatform(TestPlatforms.Android, "Android does not support AIA fetching")]
+    [SkipOnPlatform(TestPlatforms.Android | TestPlatforms.Browser, "Android nor Browser does not support AIA fetching")]
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography/tests/X509Certificates/RevocationTests/AiaTests.cs:12]: Assuming we allow multiple SkipOnPlatform, I'd leave it as two things (browser doesn't support X.509, Android doesn't support AIA)

Comment (vcsjones) [reply src/libraries/System.Security.Cryptography/tests/X509Certificates/RevocationTests/AiaTests.cs]: > Assuming we allow multiple SkipOnPlatform

Seems that we have it in many places:

https://github.com/dotnet/runtime/blob/8792a67f1c7340558eea05701c8a92120c5ca87b/src/libraries/Common/tests/StreamConformanceTests/System/IO/StreamConformanceTests.cs#L1634-L1637

**Review by bartonjs — APPROVED** [review]

**Comment by vcsjones** [comment]
WASM and outerloop are green. `extra-platforms` had no failures in System.Security and this is a test only change, so merging.

**Pull request merged**